src/arrayMethodJoin.js

```diff
@@ -4,9 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== null && this[i] !== undefined) {
```

Comment:
Replace with `this[i]`, it will be the same as `this[i] !== null && this[i] !== undefined`